umbra-js/src/utils/utils.ts

```diff
@@ -242,18 +241,6 @@ export async function lookupRecipient(
   return { spendingPublicKey: publicKey, viewingPublicKey: publicKey };
 }
 
-/**
- * @notice Creates and returns a contract instance
- * @param address Contract address
- * @param abi Contract ABI
- * @param provider ethers provider instance
- */
-export function createContract(address: string, abi: ContractInterface, provider: EthersProvider) {
-  // Use signer if available, otherwise use provider
-  const signer = provider.getSigner();
-  return new Contract(address, abi, signer || provider);
```

Comment:
@apbendi the problem you were seeing is because previously the provider that was passed in to this method was the provider from the user's wallet, so it had a valid signer attached. Now we use a regular RPC URL, and `getSigner` always returns a non-null value even though there's not actually a valid signer attached to e.g. an infura URL, and I couldn't find a good way to distinguish a valid from invalid signer on the returned signer object.

We don't need this method anymore anyway, so just got rid of it and replaced it with direct contract instantiations